crypto/pkcs11key/key.go

```diff
@@ -76,11 +111,15 @@ type Key struct {
 	pin string
 
 	// The public key corresponding to the private key.
-	publicKey rsa.PublicKey
+	rsaPublicKey rsa.PublicKey
```

Comment:
I've tried this and couldn't get it to work. Once you assign it to a crypto.PublicKey, crypto.x509.signingParamsForPublicKey() fails.

The problem also appears when changing the return types of the getPublicKey functions.